spec.html

```diff
@@ -46608,7 +46608,7 @@ <h1>%IteratorHelperPrototype%.return ( )</h1>
             1. If _O_.[[GeneratorState]] is ~suspended-start~, then
               1. Set _O_.[[GeneratorState]] to ~completed~.
               1. NOTE: Once a generator enters the completed state it never leaves it and its associated execution context is never resumed. Any execution state associated with _O_ can be discarded at this point.
-              1. Perform ? IteratorClose(_O_.[[UnderlyingIterator]], ReturnCompletion(*undefined*)).
+              1. Perform ? IteratorClose(_O_.[[UnderlyingIterator]], NormalCompletion(~unused~)).
```

Comment:
Why is it `~unused~`? Isn't this the actual return from the built-in function? It looks like `[[Call]]` and `BuiltinCallOrConstruct` just pass this along.

Reply:
No. Built-in (non-generator) functions do not use return completions. The return completions you get from evaluating ES functions are transformed to normal completions in step 8 of [`[[Call]]`](https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ecmascript-function-objects-call-thisargument-argumentslist) for ES functions; that's the place that all return completions ultimately end up. Returning a return completion here would propagate to the code which called `.return`, which is not right.